tool/tgz/reader.go

```diff
@@ -0,0 +1,56 @@
+package tgz
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+)
+
+// Reader includes a tar reader and its underlying readers.
+type Reader struct {
+	*tar.Reader
+	g *gzip.Reader
+	f *os.File
+}
+
+// OpenFile returns a Reader
+func OpenFile(fn string) (*Reader, error) {
+	f, e := os.Open(fn)
+	if e != nil {
+		return nil, e
+	}
+
+	r, e := NewReader(f)
+	if e != nil {
+		return nil, e
+	}
+
+	r.f = f
+	return r, nil
+}
+
+// NewReader wraps an io.Reader into a Reader.
+func NewReader(r io.Reader) (*Reader, error) {
```

Comment:
Is `NewReader` only used in package `tgz`? If so we should not expose it.

Reply:
I am working on make `ImageNetLoader` call `tgz.NewReader`. It should be used with the MNIST loader (the loading .tgz version) too.